### PR TITLE
Anko - Change Search screen from xml to anko layout

### PR DESCRIPTION
**Pull request by at-cuongcao**
#### Related issues (Required)
- Task #112 

#### Description (Required)
- [x] Change xml to anko layout

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+            override fun onItemClick(location: MyLocation) {
+                // TODO: 28/11/2017
+                // Dummy data
+                Log.i("tag11", location.name)
+            }
+        })
+        searchActivityUI = SearchActivityUI(adapter)
+        searchActivityUI.setContentView(this)
+    }
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 2]
```diff
@@ -1,10 +1,65 @@
...
+            locations.add(MyLocation("aaa4", "sss4", "Binh Dao",
+                    "Binh Dao Thang Binh"))
+            locations.add(MyLocation("aaa5", "sss5", "Binh Dao",
+                    "Binh Dao Thang Binh"))
+            locations.add(MyLocation("aaa6", "sss6", "Binh Dao",
+                    "Binh Dao Thang Binh"))
+            adapter.notifyDataSetChanged()
+        }
+    }
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 3]
```diff
@@ -1,10 +1,65 @@
...
+    fun getCurrentLocation() {
+        // Init later.
+        toast(R.string.coming_soon)
+    }
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING: Public functions require documentation.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        internal const val ID_TV_LOCATION_FORMAT_ADDRESS = 2205
+    }
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
+                false)).tag as LocationViewHolder
+    }
+
+    /**
+     * View holder of RecyclerView's item.
+     */
+    inner class LocationViewHolder(itemView: View, private val imgLocationIcon: ImageView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                                   private val tvLocationName: TextView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder {
+        return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            val view = with(ui) {
+                relativeLayout {
+                    lparams(matchParent, wrapContent)
+                    backgroundColor = Color.WHITE
+                    view {
+                        id = ID_VIEW_BREAK_LINE
+                        backgroundResource = R.color.colorSearchScreenBackground
+                    }.lparams {
+                        width = matchParent
+                        height = dip(0.5f)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+                        bottomMargin = dip(2)
+                        topMargin = dip(2)
+                        leftMargin = dip(64)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+                    }
+
+                    imgLocationIcon = imageView {
+                        id = ID_IMG_LOCATION_ICON
+                    }.lparams {
+                        width = wrapContent
+                        height = wrapContent
+                        margin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+                    tvLocationName = textView {
+                        id = ID_TV_LOCATION_NAME
+                        singleLine = true
+                        textSizeDimen = R.dimen.search_screen_text_size
+                    }.lparams {
+                        width = wrapContent
+                        height = wrapContent
+                        below(ID_VIEW_BREAK_LINE)
+                        leftMargin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+                    tvLocationAddress = textView {
+                        id = ID_TV_LOCATION_FORMAT_ADDRESS
+                        singleLine = true
+                        textSizeDimen = R.dimen.search_screen_text_size
+                    }.lparams {
+                        width = wrapContent
+                        height = wrapContent
+                        below(ID_TV_LOCATION_NAME)
+                        leftMargin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 13]
```diff
@@ -1,8 +1,178 @@
...
+                    imageButton {
+                        id = ID_IMG_BTN_BACK
+                        backgroundResource = R.drawable.ic_back_icon_button
+                        onClick {
+                            owner.onBackPressed()
+                        }
+                    }.lparams(wrapContent, wrapContent)
+
+                    edtLocation = editText {
+                        padding = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 14]
```diff
@@ -1,8 +1,178 @@
...
+                            override fun afterTextChanged(editable: Editable) {
+                                if (editable.toString().trim().isNotEmpty()) {
+                                    owner.searchLocations(editable.toString().trim())
+                                }
+                            }
+                        })
+                    }.lparams {
+                        width = matchParent
+                        height = matchParent
+                        leftMargin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 15]
```diff
@@ -1,8 +1,178 @@
...
+                            owner.getCurrentLocation()
+                        }
+
+                        imageView {
+                            id = ID_IMG_YOUR_LOCATION_ICON
+                            backgroundResource = R.drawable.ic_my_location
+                        }.lparams {
+                            width = wrapContent
+                            height = wrapContent
+                            margin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 16]
```diff
@@ -1,8 +1,178 @@
...
+                        }
+
+                        textView {
+                            padding = dip(5)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 17]
```diff
@@ -1,8 +1,178 @@
...
+                            textSizeDimen = R.dimen.search_screen_text_size
+                            textResource = R.string.your_location
+                        }.lparams {
+                            width = matchParent
+                            height = wrapContent
+                            margin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 18]
```diff
@@ -1,8 +1,178 @@
...
+                            rightOf(ID_IMG_YOUR_LOCATION_ICON)
+                        }
+
+                        view {
+                            backgroundResource = R.color.colorSearchScreenBackground
+                        }.lparams {
+                            width = matchParent
+                            height = dip(0.5f)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 19]
```diff
@@ -1,8 +1,178 @@
...
+                            leftMargin = dip(64)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 20]
```diff
@@ -1,8 +1,178 @@
...
+                            owner.chooseOnMap()
+                        }
+
+                        imageView {
+                            id = ID_IMG_CHOOSE_ON_MAP_ICON
+                            backgroundResource = R.drawable.ic_choose_on_map
+                        }.lparams {
+                            width = wrapContent
+                            height = wrapContent
+                            margin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 21]
```diff
@@ -1,8 +1,178 @@
...
+                        }
+
+                        textView {
+                            padding = dip(5)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 22]
```diff
@@ -1,8 +1,178 @@
...
+                            textSizeDimen = R.dimen.search_screen_text_size
+                            textResource = R.string.choose_on_map
+                        }.lparams {
+                            width = matchParent
+                            height = wrapContent
+                            margin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 23]
```diff
@@ -1,8 +1,178 @@
...
+
+                    recyclerView {
+                        id = ID_RECYCLER_VIEW_LOCATIONS
+                        layoutManager = LinearLayoutManager(context)
+                        adapter = locationAdapter
+                    }.lparams(matchParent, matchParent)
+                }.lparams {
+                    width = matchParent
+                    height = matchParent
+                    topMargin = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 24]
```diff
@@ -1,8 +1,178 @@
...
+                    below(ID_LL_HEADER_LAYOUT)
+                }
+            }.lparams {
+                width = matchParent
+                height = matchParent
+                padding = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 25]
```diff
@@ -1,10 +1,74 @@
...
+     * Get current location.
+     */
+    fun getCurrentLocation() {
+        // Init later.
+        toast(R.string.coming_soon)
+    }
+
+    /**
+     * Choose location on map.
+     */
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 26]
```diff
@@ -0,0 +1,151 @@
...
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder {
+        try {
+            return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
+                    false)).tag as LocationViewHolder
+        } catch (e: ClassCastException) {
+            throw RuntimeException(e.message)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: ERROR: Thrown exception is too generic. Prefer throwing project specific exceptions to handle error cases.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 27]
```diff
@@ -0,0 +1,151 @@
...
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder {
+        try {
+            return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 28]
```diff
@@ -0,0 +1,147 @@
...
+        private const val DEFAULT_PADDING_MARGIN = 10
+    }
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 29]
```diff
@@ -0,0 +1,147 @@
...
+            return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
+                    false)).tag as? LocationViewHolder
+    }
+
+    /**
+     * View holder of RecyclerView's item.
+     */
+    inner class LocationViewHolder(itemView: View,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 30]
```diff
@@ -0,0 +1,147 @@
...
+                                   private val imgLocationIcon: ImageView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 31]
```diff
@@ -0,0 +1,147 @@
...
+                                   private val tvLocationName: TextView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 32]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
+            return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 33]
```diff
@@ -0,0 +1,142 @@
...
+                    false)).tag as? LocationViewHolder
+    }
+
+    /**
+     * View holder of RecyclerView's item.
+     */
+    inner class LocationViewHolder(itemView: View,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 34]
```diff
@@ -0,0 +1,142 @@
...
+                                   private val imgLocationIcon: ImageView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 35]
```diff
@@ -0,0 +1,142 @@
...
+                                   private val tvLocationName: TextView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 36]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
+            return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context, parent,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 37]
```diff
@@ -0,0 +1,142 @@
...
+import android.widget.TextView
+import org.jetbrains.anko.*
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.MyLocation
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by cuongcaov on 28/11/2017
+ */
+class LocationAdapter(val locations: MutableList<MyLocation>, val listener: OnItemClick)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: remove  `interface` and use `lambda` instead

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 38]
```diff
@@ -0,0 +1,142 @@
...
+        override fun createView(ui: AnkoContext<ViewGroup>): View {
+            val view = with(ui) {
+                relativeLayout {
+                    lparams(matchParent, wrapContent)
+                    backgroundColor = Color.WHITE
+                    view {
+                        id = ID_VIEW_BREAK_LINE
+                        backgroundResource = R.color.colorSearchScreenBackground
+                    }.lparams {
+                        width = matchParent
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: width and height set on `lparams(width, height)`

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 39]
```diff
@@ -1,8 +1,170 @@
...
+                        }
+                    }.lparams(wrapContent, wrapContent)
+
+                    edtLocation = editText {
+                        padding = dimen(R.dimen.default_padding_margin)
+                        backgroundColor = Color.WHITE
+                        singleLine = true
+                        hintResource = R.string.search_location_enter_a_place
+                        textSizeDimen = R.dimen.search_screen_text_size
+                        addTextChangedListener(object : TextChangeListener {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: please use lambda syntax

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 40]
```diff
@@ -1,8 +1,170 @@
...
+                        })
+                    }.lparams {
+                        width = matchParent
+                        height = matchParent
+                        leftMargin = dimen(R.dimen.default_padding_margin)
+                    }
+                }.lparams(matchParent, wrapContent)
+
+                verticalLayout {
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: remove this line

**Comment by at-cuongcao** [comment]
Em close PR này để chuyển qua PR 126.

**Pull request closed without merging**